drivers/char/vhm/vhm_dev.c

```diff
@@ -144,20 +144,24 @@ static ssize_t vhm_dev_write(struct file *filep, const char *buffer,
 
 static void update_assigned_vf_state(uint16_t bdf, bool is_assigned)
 {
+	struct pci_bus *bus = NULL;
 	struct pci_dev *dev = NULL;
 
-	dev = pci_get_slot(pci_find_bus(0, PCI_BUS_NUM(bdf)),
-			(bdf & 0xFF));
-
-	if (dev) {
-		if (dev->is_virtfn) {
-			if (is_assigned)
-				pci_set_dev_assigned(dev);
-			else
-				pci_clear_dev_assigned(dev);
+	bus = pci_find_bus(0, PCI_BUS_NUM(bdf));
+	if (bus != NULL) {
+		dev = pci_get_slot(pci_find_bus(0, PCI_BUS_NUM(bdf)),
```

Comment:
Why do "pci_find_bus(0, PCI_BUS_NUM(bdf)" again? You already get the bus, no?

Reply:
thank you for this catch, have updated. @shuox .